### DISCUSSION TITLE
Can a debloated stock os be as clean as a custom os?

### DISCUSSION THREAD

**Discussion by dudicoco** [Universal-Debloater-Alliance/universal-android-debloater-next-generation#766] [Q&A]
Hi,

Would a desolated stock os can give us the same results as using a custom os such as lineage os?
Would there still be stock os processes running in the background or anything like that?

**Comment by MalcolmReed-ent** [comment]
Sure, a debloated stock OS will definitely be an improvement over the factory install with all its original apps and programs. It'll run smoother and be less cluttered. But when you compare it to custom ROMs like DivestOS, LineageOS, or GrapheneOS, it's not really in the same league.

Even after debloating, you're still stuck with a lot of Google-based  and Carrier-based stuff under the hood. If you're okay with potentially breaking some features to get rid of bloat, you might as well go all the way and root your phone to install a custom OS.

That way, you get a much cleaner system from the ground up, without having to work around the limitations of the stock ROM.

But I might be in the same pool as you, trying to work with a non-rootable phone and make it function better.

**Reply by dudicoco** [reply]
Thanks for the info.

> you're still stuck with a lot of Google-based and Carrier-based stuff under the hood.

Can you please be more specific about this? In a pixel for example (non-carrier), if you debloat play services, what are we still stock with?

**Reply by MalcolmReed-ent** [reply]
I mean if your rocking a pixel non-carrier you could definitely root it but if your rolling with the default ROM and after removing google play services your still stuck with the rest of the google suite if you didn't remove it, your still stuck with a hodgepodge of stuff.

For example on my phone these are some of the stuff left over that cant be disabled per se without boot loops occurring and broken stuff like

- com.google.android.webview
- com.android.wifi.resources
- com.google.cellbroadcastuiresoverlay
- com.android.android.networkstack.*
- com.google.android.overlay.*
- com.google.android.overlay.modules.*

Ultimately, at the end of the day when running stock android there's a guarantee there is some google stuff baked into it since they bought Android/ASOP and have been shipping it in practically every android phone and since its google its possibly data-mining something nor everything and monitoring whatever they want.
